### ISSUE TITLE
OSK is too small on Pro 5 in landscape mode

### ISSUE THREAD

**Issue by TartanSpartan** [ubports/ubuntu-touch#1083]
- Device: turbo (possibly just this device)
- Channel: any
- Build: any

### Steps to reproduce
Pull up the OSK to enter text in landscape mode.

### Expected behaviour
The OSK should have nice big user-friendly buttons.

### Actual behaviour
It doesn't, and hasn't since IIRC OTA-4 or so. Arale shows the correct behaviour, both in landscape and portrait mode. But turbo only shows it in portrait. This problem, combined with other bugs such as https://github.com/ubports/ubuntu-touch/issues/1030 and https://github.com/ubports/ubuntu-touch/issues/1038, makes typing in landscape more of a pain than it should be.

### Logfiles and additional information
I can follow up with screenshots next time I'm at my laptop.

**Comment by kugiigi** [comment]
This is also true on Nexus 7 2013 on its default scaling.
I found out that the height of OSK was reduced for tablets/desktops in a [commit](https://github.com/ubports/keyboard-component/commit/7145925b45ef909e5d95cbe7da59ba15d1a41445#diff-0f7eefe813944a636269157eb4f42f12).

@TartanSpartan Can you perhaps post a screenshot? I'm thinking maybe the PRO 5 is considered as a tablet based on its width/height in grid unit.

![screenshot20190603_204940290](https://user-images.githubusercontent.com/29456786/58865909-b4162c80-86b7-11e9-927f-14af79575976.png)
![screenshot20190603_205024235](https://user-images.githubusercontent.com/29456786/58865914-b5475980-86b7-11e9-8d1f-875dfad700d5.png)

**Comment by TartanSpartan** [comment]
I will post a screenshot indeed (better late than never!). What you say makes sense. Just looking at the N7's screenshot, the buttons indeed appear about proportionately the same size as on my Pro 5. I would also confirm this on my M10, however I have it locked on OTA3/Vivid for the time being.

![photo_2019-06-04_14-41-32 (2)](https://user-images.githubusercontent.com/39737544/58883810-116da600-86d7-11e9-8171-704eb8bf85b0.jpg)

Here we see the buttons in portrait are nice and chunky- nothing wrong there.

![photo_2019-06-04_14-41-32](https://user-images.githubusercontent.com/39737544/58883812-116da600-86d7-11e9-9637-37c2c8820dba.jpg)

Here we see the buttons in landscape are too small, especially for people with big fingers.

**Comment by kugiigi** [comment]
That definitely looks like the screen percentage for tablets. Phones are set to 49% on landscape while tablets are set to 34%. Tablets are identified when `width >= units.gu(90)` which I assume PRO 5 passes.

**Comment by UniversalSuperBox** [comment]
Aha, that's a really good find. Unity8 also has a configuration identifier for tablets set in `device.conf` or `DeviceConfiguration.qml` (in the case of older devices). This is something I need to write some documentation on. This property is stored as `usageScenario` by the OrientedShell.

Does the keyboard do its own determination of whether it's on a phone or tablet, @kugiigi?

**Comment by kugiigi** [comment]
Yes, it has its own detection logic which is actually too simple to cover many scenarios.
The logic is located here (`tablet` - booelan property of `fulllScreenItem`)
https://github.com/ubports/keyboard-component/blob/xenial/qml/Keyboard.qml

**Comment by UniversalSuperBox** [comment]
Alright, all of the device detection and its related configuration files can be found in [deviceconfigparser.cpp](https://github.com/ubports/unity8/blob/xenial_-_edge/plugins/Utils/deviceconfigparser.cpp) and [DeviceConfiguration.qml](https://github.com/ubports/unity8/blob/xenial_-_edge/qml/DeviceConfiguration.qml) within Unity8. @dobey, do you think it would be sound to split this off into its own little project and package so the logic can be shared? Maybe a dbus API?
